src/raftstore/store/peer_storage.rs

```diff
@@ -949,11 +988,16 @@ impl PeerStorage {
         }
 
         if ctx.raft_state != self.raft_state {
-            try!(ctx.save_raft_to(&self.engine, &mut ready_ctx.wb));
+            try!(ctx.save_raft_to(&mut ready_ctx.raft_wb));
+            if has_apply_snapshot {
+                // in case of restart happen when we just write region state to Applying,
+                // but not write raft_local_state to raft rocksdb in time.
+                try!(ctx.save_raft_to(&mut ready_ctx.wb));
```

Comment:
not only the function name, the backend key name in the rocksdb should be different too.